tests/test_utils.py

```
@@ -0,0 +1,65 @@
+
+def test_get_job_key_normalizes_strings()->None:
+    """Test that it strips whitespace and handles casing."""
+    job = {
+        'company': '  ACME CORP',
+        'title': 'DevOps Engineer',
+        'url': 'HTTP://LINK.COM'
+    }
+    result = get_job_key(job)
+    assert result == "acme corp|devops engineer|http://link.com"
+
+@pytest.mark.parametrize("job_input, expected_key", [
```

Ruff (0.15.5)

[warning] Wrong type passed to first argument of `pytest.mark.parametrize`; expected `tuple`

Use a `tuple` for the first argument

(PT006)